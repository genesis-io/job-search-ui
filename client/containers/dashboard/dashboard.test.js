import React from 'react';
import 'raf/polyfill';
import { createMockStore } from 'redux-test-utils';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history';

import Dashboard from '../../containers/dashboard/dashboard';

let component;
const testState = {
  userProfile: {
    user: {},
    isAuth: true,
  },
};
const store = createMockStore(testState);
const context = {
  store,
};

const wrappedComponent = () => {
  if (!component) {
    const history = createMemoryHistory('/');
    component = shallow(<Dashboard history={history} />, { context }).dive({ context });
  }
  return component;
}

beforeEach(() => {
  component = undefined;
});

test('should render wrapped component', () => {
  const wrapper = wrappedComponent();
  expect(wrapper).toMatchSnapshot();
});
