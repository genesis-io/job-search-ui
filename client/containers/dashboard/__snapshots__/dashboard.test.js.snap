// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render wrapped component 1`] = `
<Dashboard
  dispatch={[Function]}
  history={
    Object {
      "action": "PUSH",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "q5ldao",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        Object {
          "hash": "",
          "key": "nt3yps",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 1,
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "nt3yps",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  user={
    Object {
      "isAuth": true,
      "user": Object {},
    }
  }
/>
`;
